components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -1034,7 +1034,14 @@ public boolean testRead(String[] args)
 
     // initialize reader
     long s = System.currentTimeMillis();
-    reader.setId(id);
+    try {
+      reader.setId(id);
+    } catch (FormatException exc) {
+      reader.close();
```

Comment:
I'd propose leaving the change (though you could catch most exceptions and call `close()` I'd think), but that we also look at the wider proposal as well.